### PR TITLE
Client app

### PR DESCRIPTION
**Pull request by arextar**
This adds some code for the client app. `index.js` starts a server that listens on port `4444` and serves the ui code. The `api` folder contains example code for APIs being called through socket.io. You can ignore the .gitignore commits; that was a mistake and I wasn't quite able to get it straightened out.

### CODE REVIEW

File: app/api/achievements.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+var mongoose = require('mongoose');
```
Comment (jeremyckahn) [inline app/api/achievements.js:1]: We need to have a top-level `package.json` file that references Mongoose as a dependency if we are requiring it here.  Please add that to this Pull Request.  You can fill in minimal metadata for `package.json`, just the bare minimum for this dependency is fine.

File: app/api/index.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+var methods = {};
+
+function add_methods (name) {
+  var modul = require('./' + name);
```
Comment (jeremyckahn) [inline app/api/index.js]: Please change this variable name to `module` here and below.

File: app/index.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+///      configured through environment variables if that makes it easier.
+
+// Todo: Listen to MongoDB (not sure how it will be set up, once that's figured out we can listen for it here)
+// require('mongoose')
+
+var express = require('express');
+var http = require('http');
+var app = exports.app = express();
+var server = http.createServer(app);
+var io = exports.io = require('socket.io').listen(server);
```
Comment (jeremyckahn) [inline app/index.js:10]: As before, please create a `package.json` file and add Express and Socket.IO as dependencies.

File: app/index.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
+// Note: Ports, directories (like the location of the UI files), and such can be
+///      configured through environment variables if that makes it easier.
+
+// Todo: Listen to MongoDB (not sure how it will be set up, once that's figured out we can listen for it here)
+// require('mongoose')
```
Comment (jeremyckahn) [inline app/index.js]: Please remove this commented `require`.  Generally, I would prefer to omit commented code like this, at least in the `master` branch.

**Comment by arextar** [comment]
Made the changes you requested.

**Comment by jeremyckahn** [comment]
Overall this is solid, definitely a good start on the Node app.  Great work on this!  I left a few notes inline that need to be addressed before this can be merged - the main issue being that we need to reference the dependencies with a `package.json` file.

The Node app will need to be documented, but that can wait until after this is merged.  A wiki page is fine, but we should at least reference the URL of the corresponding wiki page in `/app/README.md`.

**Comment by jeremyckahn** [comment]
Faster than lightning.  Thanks for making the changes so quickly!

**Pull request merged**